code/States/Gamemodes/BaseGamemode.cs

```diff
@@ -62,6 +67,12 @@ public abstract partial class BaseGamemode : BaseState
 
 	protected override void Enter( bool forced, params object[] parameters )
 	{
+		if ( !IsServer )
+		{
+			base.Enter( forced, parameters );
+			return;
+		}
+
 		if ( GameConfig.TerrainFile != string.Empty )
```

Comment:
Perhaps we should move the terrain-related initialization out to another function in the future. Fine for now though.

Reply:
Yeah I can move it to a virtual method like `SetupTerrain` so other game modes can override it